sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalFileSystem.java

```diff
@@ -143,4 +165,47 @@ protected void delete(Collection<LocalResourceId> resourceIds) throws IOExceptio
       }
     }
   }
+
+  private MatchResult matchOne(String spec) throws IOException {
+    File file = Paths.get(spec).toFile();
+
+    File parent = file.getAbsoluteFile().getParentFile();
+    if (!parent.exists()) {
+      return MatchResult.create(Status.NOT_FOUND, EMPTY_METADATA);
+    }
+
+    // Method getAbsolutePath() on Windows platform may return something like
+    // "c:\temp\file.txt". FileSystem.getPathMatcher() call below will treat
+    // '\' (backslash) as an escape character, instead of a directory
+    // separator. Replacing backslash with double-backslash solves the problem.
+    // We perform the replacement on all platforms, even those that allow
+    // backslash as a part of the filename, because Globs.toRegexPattern will
+    // eat one backslash.
+    String pathToMatch = file.getAbsolutePath().replaceAll(Matcher.quoteReplacement("\\"),
+        Matcher.quoteReplacement("\\\\"));
+
+    final PathMatcher matcher =
+        java.nio.file.FileSystems.getDefault().getPathMatcher("glob:" + pathToMatch);
+
+    Iterable<File> files = com.google.common.io.Files.fileTreeTraverser().preOrderTraversal(parent);
```

Comment:
This seems like the wrong utility to use here, because we should be pruning the search tree as we go rather than enumerating all files. It looks like this will at least look at all files in the recursive subtree of parent, even if `spec` has a unique prefix. (like `/home/file-*.txt` will look at the entire recursive contents of `/home`, even recursing into `/home/some-other-dir`.)